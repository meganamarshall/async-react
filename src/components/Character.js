import React from 'react';
import PropTypes from 'prop-types';

function Character({ character }) {
  return (
    <section>
      <p>{character.name}</p>
      <p>{character.status}</p>
      <p>{character.species}</p>
      <img src={character.image} />
    </section>
  );
}

Character.propTypes = {
  character: PropTypes.shape({
    name: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })
};

export default Character;

