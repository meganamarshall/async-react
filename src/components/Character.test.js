import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('character component', () => {
  it('renders the character', () => {
    const character = {
      name: 'rick',
      status: 'alive',
      species: 'human',
      image: 'telkjse.jpg'
    };

    const wrapper = shallow(<Character character={character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
