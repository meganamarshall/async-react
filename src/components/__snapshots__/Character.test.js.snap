// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`character component renders the character 1`] = `
<section>
  <p>
    rick
  </p>
  <p>
    alive
  </p>
  <p>
    human
  </p>
  <p>
    telkjse.jpg
  </p>
</section>
`;
